### PR TITLE
Fix issue #2134: creating an anonymous user 

### PR DESCRIPTION
**Pull request by hmoqhim**
It seems to me issue #2134 is a bug (Parse Documentation (https://parse.com/docs/rest/guide#users-linking). Parse server should return the generated username when an anonymous user signs up. This PR is an attempt to resolve the issue.

### CODE REVIEW

File: src/RestWrite.js [hunk 1]
```diff
@@ -790,6 +791,10 @@ RestWrite.prototype.runDatabaseOperation = function() {
     .then(response => {
       response.objectId = this.data.objectId;
       response.createdAt = this.data.createdAt;
+      
+      if (this.hasUserNameGen)
```
Comment (flovilmart) [inline src/RestWrite.js]: Add curly braces around statement.

File: src/RestWrite.js [hunk 2]
```diff
@@ -351,6 +351,7 @@ RestWrite.prototype.transformUser = function() {
     if (!this.data.username) {
       if (!this.query) {
         this.data.username = cryptoUtils.randomString(25);
+        this.hasUserNameGen = true;
```
Comment (flovilmart) [inline src/RestWrite.js]: Use a more descriptive variable like 'responseShouldHaveUsername'

**Comment by flovilmart** [comment]
Thanks for the PR!
A few nits and we'll be able to merge that one.

**Comment by facebook-github-bot-8** [comment]
@hmoqhim updated the pull request.

**Pull request merged**